settings/movement.py

```diff
@@ -0,0 +1,169 @@
+from data.movement import MovementActions
+from memory.space import Allocate, Bank, Reserve, Read
+import instruction.asm as asm
+
+WALK_SPEED = 2
+SPRINT_SPEED = 3
+DASH_SPEED = 4
+
+class Movement:
+    def __init__(self):
+        import args
+        self.movement = args.movement or MovementActions.AUTO_SPRINT
+
+        if self.movement != MovementActions.ORIGINAL:
+            self.mod()
+
+    def mod(self):
+        length = 0
+        src = []
+
+        (length, src) = self.get_auto_sprint_src()
+
+        space = Allocate(Bank.F0, length, "Sprint subroutine")
+        space.write(src)
+
+        src = [
+            asm.JSL(space.start_address_snes),
+        ]
+        space = Reserve(0x04e21, 0x04e37, "auto sprint", asm.NOP())
```

Comment:
Looking at this vanilla code, I see that it starts with this:

```
C0/4E21:	ADB81E  	LDA $1EB8      (load event bits)
C0/4E24:	2902    	AND #$02       (is "disable sprinting" set?)
C0/4E26:	D00B    	BNE $4E33      (branch if so)
```

I'm not sure in what cases the "disable sprinting" flag would otherwise be set, but I think we'd want to keep that logic in place.

Reply:
this space is reserving the same as main right now - https://github.com/AtmaTek/WorldsCollide/blob/main/settings/auto_sprint.py#L35

Not sure how we want to handle that

Reply:
oh, thanks. No need to change then.